src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php

```diff
@@ -169,20 +175,6 @@ private function getBundleTemplatePaths(ContainerBuilder $container, array $conf
     {
         $bundleHierarchy = [];
         foreach ($container->getParameter('kernel.bundles_metadata') as $name => $bundle) {
-            $defaultOverrideBundlePath = $container->getParameterBag()->resolveValue($config['default_path']).'/bundles/'.$name;
-
-            if (file_exists($dir = $container->getParameter('kernel.root_dir').'/Resources/'.$name.'/views')) {
-                @trigger_error(sprintf('Templates directory "%s" is deprecated since Symfony 4.2, use "%s" instead.', $dir, $defaultOverrideBundlePath), E_USER_DEPRECATED);
-
-                $bundleHierarchy[$name][] = $dir;
-            }
-            $container->addResource(new FileExistenceResource($dir));
-
-            if (file_exists($defaultOverrideBundlePath)) {
-                $bundleHierarchy[$name][] = $defaultOverrideBundlePath;
-            }
-            $container->addResource(new FileExistenceResource($defaultOverrideBundlePath));
-
             if (file_exists($dir = $bundle['path'].'/Resources/views')) {
```

Comment:
Hi @alterphp, re-thinking this in another way... instead, what about moving this registration (as late as possible) into the compiler pass and call to `addPath` always?

I mean, G1 and G2 as before in the same place, and only G3 into the compiler pass, thus other bundles would `addPath` before G3 keeping the priority/order and also less changes than now.

Reply:
It took me some minutes to understand ... and it sounds far better than what I have done until now. I'm gonna take the time to refactor this.

Thanks for review !